### PR TITLE
Waits for panel reveal before measuring first cell for cell toolbar 

### LINKED ISSUE #13846: Cell toolbar overlaps rendered Markdown cell on initial load
## Description

When a notebook first loads, rendered markdown and the cell toolbar may overlap.

image

## Reproduce

1. Load a notebook whose first cell is rendered markdown that extends to the right edge.
2. Observe that the cell toolbar appears, overlapping the rendered markdown.
3. Click on another cell, then click the first cell.
4. Observe that the cell toolbar does not appear.

## Expected behavior

On initial load, the cell toolbar should not overlap the rendered markdown.

Alternatively, when #12223 is fixed, allow the cell toolbar to appear when expanded.

## Context

- Operating System and version: macOS 12.6.2 "Monterey"
- Browser and version: Firefox 102.7.0esr
- JupyterLab version: Tip of `master`

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>

<pre>
Paste the output from your browser Javascript console here, if applicable.

</pre>
</details>

### PR DESCRIPTION
**Pull request by JasonWeill**
## References

Fixes #13846.

## Code changes

The `reveal` promise is fulfilled when the content is "ready to be revealed," not when it has been revealed. Inserts a one frame (1000 ms / 60 fps) delay before measuring the first cell's content to determine whether to show the cell toolbar on the first cell.

## User-facing changes

If there is enough room† to display the cell toolbar, it should appear on the first cell after JupyterLab loads a new workbook. If there is not enough room,† it should not appear.

Before the change, the first cell's toolbar may overlap its contents:

cell toolbar overlapping contents

After the change, the overlap is less likely to occur.

cell toolbar not overlapping contents

† See known issues #13857, #13858, #13166, #13115, #12652

## Backwards-incompatible changes

None.

### CODE REVIEW

**Comment by andrii-i** [comment]
Reproduced the bug in head-of-master; built this locally and it fixes the bug, works and looks well

File: packages/cell-toolbar/src/celltoolbartracker.ts [hunk 1]
```diff
@@ -50,7 +50,12 @@ export class CellToolbarTracker implements IDisposable {
     this._toolbar.changed.connect(this._onToolbarChanged, this);
 
     // Only add the toolbar to the notebook's active cell (if any) once it has fully rendered and been revealed.
-    void panel.revealed.then(() => this._onActiveCellChanged(panel.content));
+    void panel.revealed.then(() => {
+      // Wait one frame (at 60 fps) for the panel to render the first cell, then display the cell toolbar on it if possible.
+      setTimeout(() => {
```
Comment (krassowski) [inline packages/cell-toolbar/src/celltoolbartracker.ts:55]: Did you consider using `requestAnimationFrame` instead?

Comment (fcollonval) [reply packages/cell-toolbar/src/celltoolbartracker.ts:55]: You may also try to wait for `Cell.ready`:

https://github.com/jupyterlab/jupyterlab/blob/98e684cfbe8982b65d9ea518ccb9801762cc8a8a/packages/cells/src/widget.ts#L398

Comment (JasonWeill) [reply packages/cell-toolbar/src/celltoolbartracker.ts:55]: I tried replacing the block above with code that listens to the `ready` promise @fcollonval mentioned:

```ts
    void panel.content.activeCell?.ready.then(() => {
      this._onActiveCellChanged(panel.content);
    });
```

I saw the cell toolbar appear overlapping content, as in the "before" screen shot in the description.

Comment (JasonWeill) [reply packages/cell-toolbar/src/celltoolbartracker.ts:55]: I also tried replacing the block above with code that calls `requestAnimationFrame` as @krassowski mentioned:

```ts
    void panel.revealed.then(() => {
      // Wait one frame for the panel to render the first cell, then display the cell toolbar on it if possible.
      requestAnimationFrame((timestamp) => {
        this._onActiveCellChanged(panel.content);
      });
    });
```

I also saw the cell toolbar appear as in the "before" screen shot, overlapping cell content.

Comment (fcollonval) [reply packages/cell-toolbar/src/celltoolbartracker.ts:55]: Thanks for testing the suggestion @JasonWeill

**Review by fcollonval — APPROVED** [review]

**Pull request merged**